modules/test/playwright/tests/portal-search-web/searchBar.spec.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Page, expect, mergeTests} from '@playwright/test';
+
+import {dataApiHelpersTest} from '../../fixtures/dataApiHelpersTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {productMenuPageTest} from '../../fixtures/productMenuPageTest';
+import {searchPageTest} from '../../fixtures/searchPageTest';
+import {usersAndOrganizationsPagesTest} from '../../fixtures/usersAndOrganizationsPagesTest';
+import {SearchPage} from '../../pages/portal-search-web/SearchPage';
+import {getRandomInt} from '../../utils/getRandomInt';
+import {pagesPagesTest} from '../layout-admin-web/fixtures/pagesPagesTest';
+
+export const test = mergeTests(
+	dataApiHelpersTest,
+	loginTest(),
+	productMenuPageTest,
+	usersAndOrganizationsPagesTest,
+	pagesPagesTest,
+	searchPageTest
+);
+
+test.describe('Searchbar directs to correct page', () => {
+	const TEST_PAGE = `Test${getRandomInt()}`;
+
+	test.beforeEach(async ({page, productMenuPage, staticPagesPage}) => {
+		await test.step('Create a new test page', async () => {
+			await productMenuPage.openProductMenuIfClosed();
+
+			await productMenuPage.goToPages();
+
+			await page.getByRole('button', {name: 'New'}).click();
+
+			await page
+				.getByRole('menuitem', {exact: true, name: 'Page'})
+				.click();
+
+			await staticPagesPage.addWidgetPage(TEST_PAGE);
+		});
+	});
+
+	test.afterEach(async ({page, productMenuPage}) => {
+		await test.step('Delete test page', async () => {
+			await page.goto('/');
+
+			await productMenuPage.openProductMenuIfClosed();
+
+			await productMenuPage.goToPages();
+
+			await page
+				.locator('li')
+				.filter({hasText: new RegExp(TEST_PAGE, 'i')})
+				.getByTitle('Open Page Options Menu')
+				.click();
+
+			await page.getByRole('menuitem', {name: 'Delete'}).click();
+
+			await page
+				.locator('.modal-dialog')
+				.getByRole('button', {name: 'Delete'})
+				.click();
+		});
+	});
```

Comment:
_⚠️ I haven't tested this myself, but referencing other places using this, this could be a better way than managing this in this test._

<details><summary>Referenced Files</summary>
<p>

- https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/tests/questions-web/questions.spec.ts
- https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/tests/frontend-data-set-views-web/pages/FDSFragmentPage.ts#L126-L128

</p>
</details> 

---

Instead of creating a test page in this test itself, have you tried using [isolatedLayoutTest](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/fixtures/isolatedLayoutTest.ts)?

In the `mergeTests` you could try:
```
export const test = mergeTests(
	...
	isolatedLayoutTest({type: 'widget'}),
	...
);
```

Then in the test you can try navigating to the page with:
```
test('retains impersonation parameter when suggestions is disabled @LPD-17509', async ({
		apiHelpers,
		layout, <-----------------
		page,
		productMenuPage,
		searchPage,
		usersAndOrganizationsPage,
	}) => {
        ...
        page.goto(`/web/guest${layout.friendlyURL}`); <-----------------
```

Reply:
Thanks @thektan for this! It really streamlined the process of making and tearing down a page!